test/jdk/java/lang/instrument/VerifyLocalVariableTableOnRetransformTest.java

```diff
@@ -247,8 +250,20 @@ private void saveMismatchedBytes(byte[] classfileBuffer) {
             ProtectionDomain    protectionDomain,
             byte[] classfileBuffer) {
 
-            System.out.println(this + ".transform() sees '" + className
-                + "' of " + classfileBuffer.length + " bytes.");
+            // String concatenation can cause LinkageError or ClassCircularityError (see JDK-8311534).
+            // Need to log it for analysis.
+            try {
+                System.out.println(this + ".transform() sees '" + className
+                    + "' of " + classfileBuffer.length + " bytes.");
+            } catch (Throwable t) {
+                // try to log, save the error for handling by main thread if the logging fails
+                try {
+                    transformerException.printStackTrace();
```

Comment:
Isn't this suppose to be `t.printStackTrace()`?

Reply:
Good catch!
Fixed.